Makefile

```diff
@@ -1,60 +1,67 @@
-VENV = venv
-PYTHON = $(VENV)/bin/python3
-PIP = $(VENV)/bin/pip
+PYTHON = python3
 
+VENV      = venv
+ACTIVATE := source $(VENV)/bin/activate
 
-$(VENV)/bin/activate: requirements.txt
-	python3 -m venv $(VENV)
-	$(PIP) install --upgrade pip
-	$(PIP) install -r requirements.txt
+DOCKER = docker
+TESTDB = solver-rewards-test-db
 
+# Write a marker .install file to indicate that the dependencies have been
+# installed.
+INST := $(VENV)/.install
+$(INST): requirements.txt
+	$(PYTHON) -m venv $(VENV)
+	$(ACTIVATE); pip install --upgrade pip
+	$(ACTIVATE); pip install -r requirements.txt
+	touch $@
 
-install:
-	make $(VENV)/bin/activate
+.PHONY: install
+install: $(INST)
 
+.PHONY: clean
 clean:
-	rm -rf __pycache__
+	rm -rf __pycache__ venv
```

Comment:
I'm not 100% sure about this, but it made sense to me.

Reply:
Seems reasonable. Although I rarely want to remove my python installation. There are some other cache files that could also get cleaned (like mypi cache and test cache but whateva)